src/segmented_buffer.rs

```diff
@@ -402,21 +477,9 @@ impl AsyncWrite
                         };
 
                         let pool = this.pool.clone();
-                        let segment_size = this.buf.segment_size;
-
-                        this.buf_fut.set(Some(Box::pin(async move {
-                            pool.pull().await.map(move |mut new_buf| {
-                                // clear the BytesMut
-                                new_buf.clear();
-                                // reset BytesMut to full size
-
-                                let cap = new_buf.capacity();
-                                if cap < segment_size {
-                                    new_buf.reserve(segment_size - cap);
-                                };
-                                new_buf
-                            })
-                        })));
+
+                        this.buf_fut
+                            .set(Some(Box::pin(async move { pool.pull().await })));
```

Comment:
The semantics ended up being super condensed 💪

Reply:
Yeh, it's nice and tidy here. Having ClearBuf on the object pool is a bit awkward as a crate user if we want to use stdlib or third party types as our segment types as we can't impl the external aync_buf_pool::ClearBuf trait directly for a non local type (like Vec, Bytes, BytesMut, etc).

But I wanted to wrap BytesMut with our own Buffer struct anyway, and we're the only users of async_buf_pool, so it works out nicely :)